src/core/Syntax.ml

```diff
@@ -422,54 +496,33 @@ struct
     | El tm when debug_mode ->
       Format.fprintf fmt "el %a" (pp_atomic env) tm
     | El tm ->
-      Format.fprintf fmt "%a" (pp env) tm
+      pp env penv fmt tm
     | TpVar ix ->
       Format.fprintf fmt "#var[%i]" ix
     | TpPrf cof ->
-      Format.fprintf fmt "[%a]" (pp env) cof
+      pp_bracketed (pp env P.isolated) fmt cof
     | TpESub (sub, tp) ->
       Format.fprintf fmt "[%a]%a"
-        (pp_sub env) sub
-        (pp_atomic_tp env) tp
+        (pp_sub env P.isolated) sub
+        (pp_tp env P.(right_of substitution)) tp
     | TpLockedPrf phi ->
       Format.fprintf fmt "locked %a"
         (pp_atomic env) phi
 
-  and pp_atomic_tp env fmt tp =
-    match tp with
-    | TpDim | TpCof | Nat | Univ ->
-      pp_tp env fmt tp
-    | _ ->
-      pp_braced (pp_tp env) fmt tp
-
   and pp_cof_split_branch env fmt (phi, tm) =
     let _x, envx = ppenv_bind env `Anon in
-    Format.fprintf fmt "@[<hv>%a =>@ %a@]" (pp env) phi (pp envx) tm
+    Format.fprintf fmt "@[<hv>%a =>@ %a@]"
+      (pp env P.(left_of double_arrow)) phi
+      (pp envx P.(right_of double_arrow)) tm
 
   and pp_tp_cof_split_branch env fmt (phi, tm) =
     let _x, envx = ppenv_bind env `Anon in
-    Format.fprintf fmt "@[<hv>%a =>@ %a@]" (pp env) phi (pp_tp envx) tm
+    Format.fprintf fmt "@[<hv>%a =>@ %a@]"
+      (pp env P.(left_of double_arrow)) phi
+      (pp_tp envx P.(right_of double_arrow)) tm
 
   and pp_atomic env fmt tm =
-    match tm with
-    | Var _ | Global _ | Pair _ | CofSplit _ | Dim0 | Dim1 | Cof (Cof.Meet [] | Cof.Join []) | CodeNat | CodeCircle | CodeUniv
-    | Zero | Base | Prf ->
-      pp env fmt tm
-    | Suc _ as tm when Option.is_some (to_numeral tm) ->
-      pp env fmt tm
-    | (SubIn tm | SubOut tm | ElIn tm | ElOut tm) when not debug_mode ->
-      pp_atomic env fmt tm
-    | _ ->
-      pp_braced (pp env) fmt tm
-
-  and pp_applications env fmt tm =
-    match tm with
-    | Ap (tm0, tm1) ->
-      Format.fprintf fmt "%a %a" (pp_applications env) tm0 (pp_atomic env) tm1
-    | (SubIn tm | SubOut tm | ElIn tm | ElOut tm) when not debug_mode ->
-      pp_applications env fmt tm
-    | _ ->
-      pp env fmt tm
+    pp env P.(right_of juxtaposition) fmt tm
```

Comment:
In the new Thought, `pp_atomic` should have been removed, but it was kept to minimize git diff. It now means "printing the term in the right of the juxtaposition operator".